docs/kms/README.md

```diff
@@ -1,20 +1,38 @@
 # KMS Quickstart Guide [![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io)
 
-KMS feature allows you to use Vault to generate and manages keys which are used by the minio server to encrypt objects.This document explains how to configure Minio with Vault as KMS.
+Minio uses a key-management-system (KMS) to support SSE-S3. If a client requests SSE-S3 or auto-encryption
+is enabled the Minio server encrypts each object with an unique object key which is protected by a master key
+managed by the KMS. Usually many/all object keys are protected by a single master key.
+
+Minio supports two different KMS concepts:
+ - External KMS:
+   Minio can be configured to use an external KMS i.e. [Hashicorp Vault](https://www.vaultproject.io/).
+   An external KMS decouples Minio as storage system from key-management. An external KMS can
+   be managed by a dedicated security team and allows to grant/deny access to (certain) objects
+   by en/disabling the corresponding master keys on demand.
+   However, an external KMS causes configuration and management overhead. 
+ - Direct KMS master keys:
+   Minio can also be configured to directly use a master key specified by the ENV. variable `MINIO_SSE_MASTER_KEY`.
+   Direct master keys are useful if the storage backend is not on the same machine as the Minio server - e.g.
+   if network drives or Minio gateway is used - and an external KMS would cause to much management overhead.  
+   Note: If the Minio server machine is ever compromised than the directly specified the master key must also be 
```

Comment:
s/compromised than the/compromised, then the/

Reply:
done